internal/connor/eq.go

```diff
@@ -43,15 +43,36 @@ func eq(condition, data any) (bool, error) {
 		}
 		return true, nil
 
+	case map[string]any:
+		d, ok := data.(map[string]any)
+		if !ok {
```

Comment:
todo: if `ok` is false here that suggests a programming error in which case we should not quietly ignore the value.  Please either remove this check, or convert it to a nil check and test it if that is what you mean to check here.

Reply:
good catch! I've changed it as suggested